### DISCUSSION TITLE
Confusion about permissions

### DISCUSSION THREAD

**Discussion by simonflick** [directus/directus#17916] [Archive]
I'm new to Directus and want to create a user role called "editor" which can only edit it's own user settings. But this leads to the following problem: editors can still see all the data of other users and edit the fields. Only when they try to save the changes, an error occurs. This error is also not translated and difficult for non-programmers to understand:
> [FORBIDDEN] You don't have permission to access this.

I could restrict the role so that they can only see their own user. But then the other users would disappear everywhere like in comments.
I would like to make things not editable or hide them instead of showing errors on save. Am I doing it wrong or is this impossible to do right now?

**Comment by david-zacharias** [comment]
I think it is not possible right now, becuase as soon as a user as premission to edit an entity the pencil icon for entering the details is unlocked in the UI. 

However, as a kind of interim solution, you could disable the "User Directory" in the [Project Settings](https://docs.directus.io/configuration/project-settings.html#modules). Editors can still access and edit their own profile via the Profil Icon on the lower left corner. But note: Disabling a module is currently affecting all users regardless of their roles, including admins.

**Reply by simonflick** [reply]
Hi David, thanks for clearing it up. I would love to see this fixed in the future. But it might work to disable the User Directory in my case.
